exchangelib/util.py

```diff
@@ -512,10 +512,15 @@ def post_ratelimited(protocol, session, url, headers, data, allow_redirects=Fals
             # Always create a dummy response for logging purposes, in case we fail in the following
             r = DummyResponse(url=url, headers={}, request_headers=headers)
             try:
-                data = data.decode('utf-8')
-                request = session.prepare_request(Request('POST', url=url, data=data, headers=headers))
-                request.headers['Content-Length'] = str(len(data.encode('utf-8')))
-                r = session.send(request,
+                data = data.encode('utf-8')
+            except UnicodeDecodeError:
+                import chardet
+                encoding_info = chardet.detect(data)
+                data = data.decode('utf-8').encode(encoding_info['encoding'])
+            try:
+                r = session.post(url=url,
```

Comment:
Why does this need to be changed to .post instead of .send?

Reply:
Actually never mind. I get it.

Reply:
Yeah, sorry! I reverted back to the method we were using before since we no longer need to manually set the content length